Sources/SystemInternals/Exports.swift

```diff
@@ -36,47 +50,129 @@ public var system_errno: CInt {
 #endif
 
 public func system_open(_ path: UnsafePointer<CChar>, _ oflag: Int32) -> CInt {
+#if os(Windows)
+  var fh: CInt = -1
+  _ = _sopen_s(&fh, path, oflag, _SH_DENYNO, _S_IREAD | _S_IWRITE)
+  return fh
+#else
   open(path, oflag)
+#endif
 }
 
+#if os(Windows)
+public func system_open(
+  _ path: UnsafePointer<CChar>, _ oflag: Int32, _ mode: CInt
+) -> CInt {
+  var fh: CInt = -1
+  _ = _sopen_s(&fh, path, oflag, _SH_DENYNO, mode)
+  return fh
+}
+#else
 public func system_open(
   _ path: UnsafePointer<CChar>, _ oflag: Int32, _ mode: mode_t
 ) -> CInt {
   open(path, oflag, mode)
 }
+#endif
 
 public func system_close(_ fd: Int32) -> Int32 {
-  close(fd)
+#if os(Windows)
+  return _close(fd)
+#else
+  return close(fd)
+#endif
 }
 
+#if os(Windows)
+public func system_read(
+  _ fd: Int32, _ buf: UnsafeMutableRawPointer!, _ nbytes: CUnsignedInt
+) -> CInt {
+  _read(fd, buf, nbytes)
+}
+#else
 public func system_read(
   _ fd: Int32, _ buf: UnsafeMutableRawPointer!, _ nbyte: Int
 ) -> Int {
   read(fd, buf, nbyte)
 }
+#endif
 
 public func system_pread(
   _ fd: Int32, _ buf: UnsafeMutableRawPointer!, _ nbyte: Int, _ offset: off_t
 ) -> Int {
+#if os(Windows)
+  let handle: intptr_t = _get_osfhandle(fd)
+  if handle == /* INVALID_HANDLE_VALUE */ -1 { return Int(EBADF) }
+
+  // NOTE: this is a non-owning handle, do *not* call CloseHandle on it
+  let hFile: HANDLE = HANDLE(bitPattern: handle)!
+
+  var ovlOverlapped: OVERLAPPED = OVERLAPPED()
+  ovlOverlapped.OffsetHigh = DWORD(UInt32(offset >> 32) & 0xffffffff)
+  ovlOverlapped.Offset = DWORD(UInt32(offset >> 0) & 0xffffffff)
+
+  var nNumberOfBytesRead: DWORD = 0
+  if !ReadFile(hFile, buf, DWORD(nbyte), &nNumberOfBytesRead, &ovlOverlapped) {
+    let _ = GetLastError()
+    // TODO(compnerd) map windows error to errno
+    return Int(-1)
+  }
+  return Int(nNumberOfBytesRead)
+#else
   pread(fd, buf, nbyte, offset)
+#endif
 }
 
 public func system_lseek(
   _ fd: Int32, _ off: off_t, _ whence: Int32
 ) -> off_t {
-  lseek(fd, off, whence)
+#if os(Windows)
+  // TODO(compnerd) should this use _lseeki64 to get larget file offsets?
+  return _lseek(fd, off, whence)
+#else
+  return lseek(fd, off, whence)
+#endif
```

Comment:
To reduce the number of `#if/#else` inside of these functions, should Windows have a `internal let lseek = _lseek`, etc?

Reply:
There is a `lseek` as well, so I don't think that we should be shadowing that.